Build failure: python311.floret

### Steps To Reproduce

This turned up in a `nixpkgs-review` check for a `python311Packages.spacy` upgrade. The floret build error is unrelated to the other upgrade.

```console
$ hydra-check python311Packages.floret
Build Status for nixpkgs.python311Packages.floret.x86_64-linux on unstable
✖ (Failed) python3.11-floret-0.10.5 from 2024-02-15 - https://hydra.nixos.org/build/249244709

Last Builds:
✖ (Failed) python3.11-floret-0.10.5 from 2024-01-25 - https://hydra.nixos.org/build/247420322
✖ (Failed) python3.11-floret-0.10.5 from 2024-01-13 - https://hydra.nixos.org/build/246273631
✔ python3.11-floret-0.10.5 from 2023-12-12 - https://hydra.nixos.org/build/243761958
✔ python3.11-floret-0.10.5 from 2023-11-22 - https://hydra.nixos.org/build/241752609
✔ python3.11-floret-0.10.4 from 2023-11-18 - https://hydra.nixos.org/build/241391704
✔ python3.11-floret-0.10.4 from 2023-10-29 - https://hydra.nixos.org/build/239672132
```

### Build log

The build runs normally until `uint64_t` isn't found.
```console
g++ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -I/nix/store/as77gyp0swpalaa91f1ck7q406gki8cd-libxcrypt-4.4.36/include -fPIC -I/nix/store/8fgxhwbd5y6z7i5i76fgi3cs2ql57i6c-python3.11-pybind11-2.11.1/lib/python3.11/site-packages/pybind11/include -I/nix/store/8fgxhwbd5y6z7i5i76fgi3cs2ql57i6c-python3.11-pybind11-2.11.1/lib/python3.11/site-packages/pybind11/include -Isrc -I/nix/store/y027d3bvlaizbri04c1bzh28hqd6lj01-python3-3.11.7/include/python3.11 -c src/args.cc -o build/temp.linux-x86_64-cpython-311/src/args.o -DVERSION_INFO=\"0.10.5\" -std=c++11 -fvisibility=hidden
src/args.cc: In member function ‘void fasttext::Args::parseArgs(const std::vector<std::__cxx11::basic_string<char> >&)’:
src/args.cc:132:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<std::__cxx11::basic_string<char> >::size_type’ {aka ‘long unsigned int’} [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wsign-compare-Wsign-compare8;;]
  132 |   for (int ai = 2; ai < args.size(); ai += 2) {
      |                    ~~~^~~~~~~~~~~~~
src/args.cc:250:19: warning: catching polymorphic type ‘class std::out_of_range’ by value [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcatch-value=-Wcatch-value=8;;]
  250 |     } catch (std::out_of_range) {
      |                   ^~~~~~~~~~~~
src/args.cc: In member function ‘int64_t fasttext::Args::getAutotuneModelSize() const’:
src/args.cc:512:3: error: ‘uint64_t’ was not declared in this scope
  512 |   uint64_t multiplier = 1;
      |   ^~~~~~~~
src/args.cc:17:1: note: ‘uint64_t’ is defined in header ‘<cstdint>’; did you forget to ‘#include <cstdint>’?
   16 | #include <unordered_map>
  +++ |+#include <cstdint>
   17 | 
src/args.cc:515:5: error: ‘multiplier’ was not declared in this scope
  515 |     multiplier = units[lastCharacter];
      |     ^~~~~~~~~~
src/args.cc:518:11: error: expected ‘;’ before ‘size’
  518 |   uint64_t size = 0;
      |           ^~~~~
      |           ;
src/args.cc:522:5: error: ‘size’ was not declared in this scope
  522 |     size = std::stol(modelSize, &nonNumericCharacter);
      |     ^~~~
src/args.cc:534:10: error: ‘size’ was not declared in this scope
  534 |   return size * multiplier;
      |          ^~~~
src/args.cc:534:17: error: ‘multiplier’ was not declared in this scope
  534 |   return size * multiplier;
      |                 ^~~~~~~~~~
error: command '/nix/store/4cjqvbp1jbkps185wl8qnbjpf8bdy8j9-gcc-wrapper-13.2.0/bin/g++' failed with exit code 1

```

### Additional context

`python311Packages.textacy` is also refusing to build due to this dependency. Patching the dependency fixes the build.

```console
❯ hydra-check python311Packages.textacy
Build Status for nixpkgs.python311Packages.textacy.x86_64-linux on unstable
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-23 - https://hydra.nixos.org/build/250600466

Last Builds:
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-21 - https://hydra.nixos.org/build/250462424
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-19 - https://hydra.nixos.org/build/250316854
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-17 - https://hydra.nixos.org/build/249822702
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-15 - https://hydra.nixos.org/build/249727724
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-14 - https://hydra.nixos.org/build/249705625
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-13 - https://hydra.nixos.org/build/249525056
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-10 - https://hydra.nixos.org/build/249279190
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-02-05 - https://hydra.nixos.org/build/248543213
✖ (Dependency failed) python3.11-textacy-0.13.0 from 2024-01-30 - https://hydra.nixos.org/build/247995932
```

PR with patch forthcoming

### Notify maintainers

@GaetanLepage 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.78, NixOS, 24.05 (Uakari), 24.05.20240221.0e74ca9`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.05"`
 - channels(sec): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/floret/cstdint.patch
+++ b/pkgs/development/python-modules/floret/cstdint.patch
@@ -1,0 +1,12 @@
+diff --git a/src/args.cc b/src/args.cc
+index a8975e81624c..99854c919341 100644
+--- a/src/args.cc
++++ b/src/args.cc
+@@ -10,6 +10,7 @@
+ 
+ #include <stdlib.h>
+ 
++#include <cstdint>
+ #include <iostream>
+ #include <stdexcept>
+ #include <string>

--- a/pkgs/development/python-modules/floret/default.nix
+++ b/pkgs/development/python-modules/floret/default.nix
@@ -23,6 +23,8 @@ buildPythonPackage rec {
     hash = "sha256-7vkw6H0ZQoHEwNusY6QWh/vPbSCdP1ZaaqABHsZH6hQ=";
   };
 
+  patches = [./cstdint.patch ];
+
   nativeBuildInputs = [
     pybind11
     setuptools

